src/ILLink.RoslynAnalyzer/ILLink.RoslynAnalyzer.csproj

```diff
@@ -9,6 +9,8 @@
     <!-- There are currently no translations, so the satellite assemblies are a waste of space. -->
     <EnableXlfLocalization>false</EnableXlfLocalization>
     <EnableDefaultEmbeddedResourceItems>false</EnableDefaultEmbeddedResourceItems>
+    <!-- Let different analyzers emit the same diagnostic id. -->
+    <NoWarn>RS1019</NoWarn>
```

Comment:
It may be worth it to just fold this into the RequiresUnreferencedCode analyzer to avoid the complication

Reply:
Folded. For this I added
```C#
private protected abstract ImmutableArray<(Action<OperationAnalysisContext> Action, OperationKind[] OperationKind)> ExtraOperationActions
```
Into `RequiresAnalyzerBase` so that any analyzer can pass the operation actions that it will register. Since we cannot pass a diagnostic id string to the AnalyzerVerifier (used by the tests) when there's more than one supported diagnostic using the same id, I had to make the diagnostic descriptors of the RUC analyzer public. I'll hold off this PR until we settle a way for generating/retrieving diagnostic descriptors.